src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -381,23 +397,23 @@ private CertificateFactory getCertFactory(String algorithm) {
 
     /**
      * Configures and returns a new {@code PEMDecoder} instance from the
-     * current instance that will use KeyFactory and CertificateFactory classes
-     * from the specified {@link Provider}.  Any errors using the
-     * {@code provider} will occur during decoding.
+     * current instance that will use {@link KeyFactory} and
+     * {@link CertificateFactory} classes from the specified {@link Provider}.
+     * Any errors using the {@code provider} will occur during decoding.
      *
      * <p>If {@code provider} is {@code null}, a new instance is returned with
      * the default provider configuration.
      *
-     * @param provider the Factory provider.
+     * @param provider the factory provider.
```

Comment:
This is a little awkward because the argument of `withFactory` is not the factory itself. Shall we rename it?

Also, can we add some more description on how this method is used? For example, suppose a provider named `P1` extends `ECPublicKey` to `P1ECPublicKey`, then users should call `withFactory(p1).decode(pem, P1ECPublicKey.class)`. I assume we don't want to do some kind of delayed provider selection to make it possible with the "original" decoder.